src/rabin.js

```diff
@@ -2,197 +2,218 @@
 A Rabin Signature JavaScript module adapted
 from: https://github.com/scrypt-sv/rabin/blob/master/rabin.py
 */
-const { toBigIntLE } = require('bigint-buffer');
+const { toBigIntLE } = require( 'bigint-buffer' );
 const { checkIfValidHexString,
-    hexStringToBigInt,
-    bigIntAbsoluteValue } = require('./utils');
+  bigIntAbsoluteValue } = require( './utils' );
 let crypto;
 try {
-    crypto = require('crypto');
-} catch (err) {
-    throw('Error: crypto support is disabled!');
+  crypto = require( 'crypto' );
+} catch ( err ) {
+  throw ( 'Error: crypto support is disabled!' );
 }
-    
-function greatestCommonDivisor(a, b) {
-    if ((typeof a !== 'bigint') || (typeof b !== 'bigint'))
-        throw "Error: Incorrect argument(s) to greatestCommonDivisor()";
-
-    a = bigIntAbsoluteValue(a);
-    b = bigIntAbsoluteValue(b);
-    if (b > a) {
-        let t = a;
-        a = b;
-        b = t;
+
+class RabinSignature {
+  static paddingBuffer = Buffer.from( '00', 'hex' );
+
+  /**
+   * RabinSignature
+   * @param {*} strong : from 1(512bit) to 6(3072bit) or more
+   */
+  constructor ( strong = 1 ) {
```

Comment:
strong -> securityLevel

Also check it's within range and report error it's not.

It can be larger than 3072, right?

constructor ( securityLevel = 1 ) {
  if (securityLevel < 1) // error
  this.securityLevel = securityLevel
}